src/osdc/ObjectCacher.cc

```diff
@@ -410,10 +410,13 @@ void ObjectCacher::Object::audit_buffers()
       ceph_assert(it->first == it->second->start());
     }
     if (it->first < offset) {
-      lderr(oc->cct) << "AUDIT FAILURE: " << it->first << " " << *it->second
-		     << " overlaps with previous bh " << *((--it)->second)
+      auto cur = it;
+      auto prev = std::prev(cur);
+      lderr(oc->cct) << "AUDIT FAILURE: " << it->first << " " << *cur->second
+		     << " overlaps with previous bh " << *(prev->second)
```

Comment:
could be more consistent on using the parenthesis here. specifically, either do not use parenthesis at all, or use them on both `cur->second` and `prev->second`.